### ISSUE TITLE
Ignore devDependencies version mismatch in bootstrap

### ISSUE THREAD

**Issue by derek-ho** [opensearch-project/OpenSearch-Dashboards#5561]
**Is your feature request related to a problem? Please describe.**

Security dashboards plugin was recently cut a ticket for a build failure: https://github.com/opensearch-project/security-dashboards-plugin/issues/1682#issuecomment-1833939778. Locally, matching the version in package.json solves this type of issue, but not sure if this is scalable long term, and especially for devDependencies matching the version should matter less?

**Describe the solution you'd like**

Ignore/skip/provide some other long term solution for dependency resolution/matching

**Describe alternatives you've considered**

**Additional context**

**Comment by derek-ho** [comment]
@kavilla / @ashwin-pc I saw that there was some discussion about this before, but I wasn't able to find a resolution - can you advise here on if there is a fix from core side or we should just try to manually match the `package.json` versions?

**Comment by abbyhu2000** [comment]
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3095 i found this RFC

**Comment by derek-ho** [comment]
Thanks, @abbyhu2000, @AMoo-Miki! Would it be feasible to fix this from dashboards side in near term, or should I still open issues in individual plugin repos to manually match the versions?

**Comment by wanglam** [comment]
@derek-ho According the [opensearch-build script](https://github.com/opensearch-project/opensearch-build/blob/main/scripts/default/opensearch-dashboards/build.sh#L95), it will remove the success build plugin folder if everything goes well. According the build log, the root cause was [dashboards-reporting](https://github.com/opensearch-project/dashboards-reporting/issues/263) build failed due to failed to run `yarn osd bootstrap`. I think we need to fix this issue first.

**Comment by ashwin-pc** [comment]
@AMoo-Miki 's PR https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5675 should allow for less restrictive version requirements.

**Comment by derek-ho** [comment]
@ashwin-pc @AMoo-Miki any updates on this issue long term? We started seeing failures on bootstrap on `main` again:
```
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning "@elastic/eslint-import-resolver-kibana > eslint-import-resolver-webpack@0.13.8" has unmet peer dependency "eslint-plugin-import@>=1.4.0".
warning " > @testing-library/react-hooks@7.0.2" has unmet peer dependency "react@>=16.9.0".
warning "@testing-library/react-hooks > react-error-boundary@3.1.4" has unmet peer dependency "react@>=16.13.1".
warning "cypress-mochawesome-reporter > mochawesome@7.1.3" has unmet peer dependency "mocha@>=7".
warning " > eslint-plugin-cypress@2.15.1" has unmet peer dependency "eslint@>= 3.2.1".
[4/4] Building fresh packages...

ERROR [single_version_dependencies] Multiple version ranges for the same dependency
      were found declared across different package.json files. Please consolidate
      those to match across all package.json files. Different versions for the
      same dependency is not supported.

      If you have questions about this please reach out to the operations team.

      The conflicting dependencies are:

        cypress
          9.5.4 => opensearch-dashboards
          ^13.6.0 => opensearch-security-dashboards
```

Any suggestions on path forward to unblock the CI/autocuts?

**Comment by derek-ho** [comment]
Update: adding `--single-version=loose` as suggested in the linked PR fixed the bootstrapping process on local CI. Opensearch-build is not passing in this param, and until it is done so autocuts will be cut. Not sure what the process forward for this issue is, but ideally this parameter shouldn't be needed for devDependencies.

**Comment by peternied** [comment]
@ashwin-pc @AMoo-Miki Could you take a look at a change in opensearch-build to make this

**Comment by AMoo-Miki** [comment]
OSD 2.12 shipped with #5675. Infra will take care of the PR on the build repo.
